webapi/src/main/scala/org/knora/webapi/slice/ontology/api/service/RestCardinalityService.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.ontology.api.service
+import zio.IO
+import zio.Task
+import zio.ZIO
+import zio.ZLayer
+import zio.macros.accessible
+
+import dsp.errors.BadRequestException.invalidQueryParamValue
+import dsp.errors.BadRequestException.missingQueryParamValue
+import dsp.errors.ForbiddenException
+import org.knora.webapi.IRI
+import org.knora.webapi.messages.admin.responder.usersmessages.UserADM
+import org.knora.webapi.messages.v2.responder.CanDoResponseV2
+import org.knora.webapi.slice.ontology.api.service.RestCardinalityService.classIriKey
+import org.knora.webapi.slice.ontology.api.service.RestCardinalityService.newCardinalityKey
+import org.knora.webapi.slice.ontology.api.service.RestCardinalityService.propertyIriKey
+import org.knora.webapi.slice.ontology.domain.model.Cardinality
+import org.knora.webapi.slice.ontology.domain.service.CardinalityService
+import org.knora.webapi.slice.ontology.domain.service.ChangeCardinalityCheckResult
+import org.knora.webapi.slice.ontology.domain.service.ChangeCardinalityCheckResult.ChangeCardinalityCheckResult
+import org.knora.webapi.slice.ontology.domain.service.OntologyRepo
+import org.knora.webapi.slice.resourceinfo.domain.InternalIri
+import org.knora.webapi.slice.resourceinfo.domain.IriConverter
+
+@accessible
+trait RestCardinalityService {
+
+  def canUpdateCardinality(
+    classIri: IRI,
+    user: UserADM,
+    propertyIri: Option[IRI],
+    newCardinality: Option[String]
+  ): Task[CanDoResponseV2] =
+    (propertyIri, newCardinality) match {
+      case (None, Some(_))                           => ZIO.fail(missingQueryParamValue(propertyIriKey))
+      case (Some(_), None)                           => ZIO.fail(missingQueryParamValue(newCardinalityKey))
+      case (None, None)                              => canReplaceCardinality(classIri, user)
+      case (Some(propertyIri), Some(newCardinality)) => canSetCardinality(classIri, propertyIri, newCardinality, user)
+    }
+
+  def canReplaceCardinality(classIri: IRI, user: UserADM): Task[CanDoResponseV2]
```

Comment:
I think this was previously (i.e. until now) the method used for updating cardinalities. So, once you introduce `canUpdateCardinality`, this one is probably no longer used?

Reply:
I did not want to introduce a breaking change so that the existing API behaviour remains in place.